src/ethereum_wallet.rs

```diff
@@ -0,0 +1,209 @@
+use crate::{geth, Seed};
+use comit::{
+    actions::ethereum::{CallContract, DeployContract},
+    asset::Erc20,
+    ethereum::{Address, ChainId, Hash, TransactionReceipt},
+};
+use reqwest::Url;
+
+#[derive(Debug, Clone)]
+pub struct Wallet {
+    private_key: clarity::PrivateKey,
+    geth_client: geth::Client,
+}
+
+impl Wallet {
+    pub fn new(seed: Seed, url: Url) -> anyhow::Result<Self> {
+        let private_key = clarity::PrivateKey::from_slice(&seed.into_inner())
+            .map_err(|_| anyhow::anyhow!("Failed to derive private key from slice"))?;
+
+        let geth_client = geth::Client::new(url);
+
+        Ok(Self {
+            private_key,
+            geth_client,
+        })
+    }
+
+    pub fn new_from_private_key(private_key: clarity::PrivateKey, url: Url) -> Self {
```

Comment:
if that's only used in test it'd be good to have it behind `#[cfg(test)]` if possible